services/subscriber/http.go

```diff
@@ -1,39 +1,88 @@
 package subscriber
 
 import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
 	"time"
 
-	"github.com/influxdata/influxdb/client/v2"
 	"github.com/influxdata/influxdb/coordinator"
 )
 
 // HTTP supports writing points over HTTP using the line protocol.
 type HTTP struct {
-	c client.Client
+	url         *url.URL
+	accessToken string
+	httpClient  *http.Client
 }
 
 // NewHTTP returns a new HTTP points writer with default options.
 func NewHTTP(addr string, timeout time.Duration) (*HTTP, error) {
-	conf := client.HTTPConfig{
-		Addr:    addr,
-		Timeout: timeout,
-	}
-	c, err := client.NewHTTPClient(conf)
+	u, err := url.Parse(addr)
 	if err != nil {
 		return nil, err
+	} else if u.Scheme != "http" && u.Scheme != "https" {
+		return nil, fmt.Errorf("Unsupported protocol scheme: %s, your address must start with http:// or https://", u.Scheme)
+	}
+	// Grab username from URL as token.
+	var token string
+	if u.User != nil {
```

Comment:
I don't really have a problem with this, but I feel like in other areas of the project we have used, or would use, a URL parameter to provide a token, rather than the username prefix to a URL's hostname.

Maybe somebody else has some stronger opinions on that one way or the other.